ArduCopter/ekf_check.cpp

```diff
@@ -72,8 +77,10 @@ void Copter::ekf_check()
             // if compass is flagged as bad and the counter reaches zero then clear flag
             if (ekf_check_state.bad_variance && ekf_check_state.fail_count == 0) {
                 ekf_check_state.bad_variance = false;
-                // log recovery in the dataflash
-                Log_Write_Error(ERROR_SUBSYSTEM_EKFCHECK, ERROR_CODE_EKFCHECK_VARIANCE_CLEARED);
+                if (should_act) {
+                    // log recovery in the dataflash
+                    Log_Write_Error(ERROR_SUBSYSTEM_EKFCHECK, ERROR_CODE_EKFCHECK_VARIANCE_CLEARED);
+                }
                 // clear failsafe
                 failsafe_ekf_off_event();
```

Comment:
@OXINARF You're advocating for doing the logging here too, then?

I think the reason I did it this way is to avoid confusion when looking at logs where you see failsafes cleared but not set.  I think you might even see one of those per boot if you're logging-before-arming.